### DISCUSSION TITLE
Feature Request: HTTP API for managing task hubs

### DISCUSSION THREAD

**Discussion by sebastianburckhardt** [Azure/azure-functions-durable-extension#2164] [Ideas]
Currently, task hubs are automatically created when they are used the first time, and the only way to delete them is to delete the containing storage or to manually delete certain objects in storage. This is very brittle, backend-dependent and difficult to discover.

Perhaps it would make sense to offer backend-independent HTTP APIs to create or delete task hubs.

**Comment by davidmrdavid** [comment]
@sebastianburckhardt: Just triple checking if you're aware of the `func durable` CLI options? The `func` CLI / core-tools has a durable option that exposes various management APIs, including taskhub deletion. I only point this out because I just recently learned about it, so perhaps I'm not the only one. See below:

```
PS C:\Users\dajusto.REDMOND> func durable

                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Usage: func durable [context] <action> [-/--options]

Actions:
delete-task-hub     Delete all storage artifacts in the durable task hub
```

func-cli aside, what behaviors are we interested in supporting here. Is it just taskhub deletion and creation or is there anything else that could be useful?

**Reply by cgillum** [reply]
The problem with the `func durable` commands is that they only work with the Azure Storage backend today. We would need to update them to also work with the alternate backends.

**Comment by cgillum** [comment]
This is an issue we discovered while developing tests for the Java SDK and work was done to support it.

Here are the client APIs in the Java SDK that can be used to create and delete task hubs: https://github.com/microsoft/durabletask-java/blob/ae18755be8c4d028074773606f0ffa9ab01715ad/sdk/src/main/java/com/microsoft/durabletask/DurableTaskClient.java#L74-L76.

Here is the implementation of these APIs in the Durable Task extension: https://github.com/Azure/azure-functions-durable-extension/blob/d9d4f0df79d37c70b8b1b333bbaf14a9657e96a7/src/WebJobs.Extensions.DurableTask/LocalGrpcListener.cs#L137-L147

I agree it would be useful to expose these over HTTP as well. I do wonder if we need to consider some kind of authorization policy for these, though, since delete in particular is very destructive.
